File: LeetCode/JavaScript/B22-739-Daily Temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function (temperatures) {
  const answer = new Array(temperatures.length);
  answer.fill(0);
  const stack = new Array();
  for (i in temperatures) {
    while (!!stack.length && temperatures[stack[stack.length - 1]] < temperatures[i]) {
      const prev = stack.pop();
      answer[prev] = i - prev;
    }
    stack.push(i);
  }

  return answer;
};

console.log(dailyTemperatures([73, 74, 75, 71, 69, 72, 76, 73]));

File: LeetCode/5회차/B22-739-Daily Temperatures.py
from typing import List


class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        for i in range(len(temperatures)):
            while stack and temperatures[stack[-1]] < temperatures[i]:
                prev = stack.pop()
                temperatures[prev] = i - prev
            stack.append(i)

        while stack:
            temperatures[stack.pop()] = 0

        return temperatures
